text/0860-ember-data-request-service.md

```diff
@@ -0,0 +1,555 @@
+---
+stage: accepted
+start-date: 2023-11-10
+release-date: 
+release-versions:
+teams:
+  - data
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/860
+project-link:
+---
+
+<!--- 
+Directions for above: 
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+-->
+
+# EmberData | Request Service
+
+## Summary
+
+Proposes a simple abstraction over `fetch` to enable easy management of request/response
+flows. Provides associated utilities to assist in migrating to this new abstraction. Adds
+ new APIs to the Store to make use of this abstraction.
+
+## Motivation
+
+- `Serializer` lacks the context necessary to serialize/normalize data on a per-request basis
```

Comment:
I can't think of a time when I've need to do this.

Reply:
Amending requests and responses in the adapter happens quite often. In fact it turns out to be so necessary that both EmberData's default adapters do this and EmberData's core logic does this post-serialization but with-request-context in order to ensure that the cache receives enough context to correctly be patched. These special things EmberData is doing come at a high cost to flexibility and performance, and can't be rectified by user-land changes. So even if you aren't doing this, *you are* actually doing this.

I'd say every app that I've worked on has had to customize behavior in the adapter to support normalization or serialization at least once. It could be that I've hit some weird bias where in a hundred apps I haven't encountered the opposite, but it's also simply the case that more often than not APIs include meta in headers, return error responses in different formats, utilize binary formats to save bytes, or return `""` or `{}` on success when *something* more was expected or needed.